src/libs/state-history.js

```diff
@@ -1,5 +1,54 @@
-export const set = newState => { window.history.replaceState(newState, '') }
+import _ from 'lodash/fp'
+import uuid from 'uuid/v4'
 
-export const get = () => window.history.state || {}
+
+const getKey = () => {
+  const state = window.history.state
+  if (state && state.key) {
+    return state.key
+  } else {
+    const key = uuid()
+    window.history.replaceState({ key }, '')
+    return key
+  }
+}
+
+
+export const get = () => {
+  const fetched = sessionStorage.getItem(getKey())
+  if (fetched) {
+    return JSON.parse(fetched).value
```

Comment:
FYI, if these values do get large, then JSON parsing time can become significant. Just something to keep an eye on.

Reply:
Should we make `get` async?

Reply:
Sounds like a good idea

Reply:
That won't work the way you're thinking. The parsing is still a synchronous blocking operation, unless you do it in a web worker (which has all sorts of other complications)